mm2src/lp_ordermatch.rs

```diff
@@ -3577,7 +3577,28 @@ struct OrderStatusReq {
     uuid: Uuid,
 }
 
+#[derive(Serialize)]
+struct OrderForRpcWithCancellationReason<'a> {
+    #[serde(flatten)]
+    order: OrderForRpc<'a>,
+    cancellation_reason: &'a String,
```

Comment:
As per https://doc.rust-lang.org/std/str/index.html: `Unlike its String counterpart, its contents are borrowed.` I think it will be more intuitive to have the `&'a str` type here.